### PR TITLE
Various cleanup

### PR DESCRIPTION
**Pull request by acburdine**
- Remove console.log in resolver
- Ensure application instance is registered as a singleton

### CODE REVIEW

File: lib/runtime/application.ts [hunk 1]
```diff
@@ -99,7 +99,7 @@ export default class Application extends Addon {
     this.drainers = [];
 
     // Setup some helpful container shortcuts
-    this.container.register('app:main', this);
+    this.container.register('app:main', this, { singleton: true });
```
Comment (davewasmer) [inline lib/runtime/application.ts]: Good catch, and this should be `singleton: true, instantiate: false` as well.

**Comment by acburdine** [comment]
@davewasmer fixed that, also found a couple of other container refs that needed to be updated

File: app/views/error.html.ts [hunk 2]
```diff
@@ -57,6 +57,9 @@ let template = compileTemplate(`
 export default class ErrorView extends View {
 
   async render(action: Action, response: ServerResponse, error: any, options: RenderOptions) {
+    // Action must be defined, or lodash's template will throw an error
+    error.action = error.action || null;
```
Comment (davewasmer) [inline app/views/error.html.ts]: Actually, if we keep the `data.error` prefixes, and use the `variable` option when compiling the template, we can ensure no `with` statement is used, and therefore no reference errors will be thrown:

```
template = compileTemplate(..., { variable: 'data' });
```

Comment (acburdine) [inline app/views/error.html.ts]: ah, didn't know that was a thing. Will fix asap :)

**Comment by acburdine** [comment]
Should be fixed now.

**Pull request merged**